java/vector/src/main/java/org/apache/arrow/vector/complex/impl/ComplexWriterImpl.java

```diff
@@ -139,17 +139,22 @@ public MapWriter directMap(){
   }
 
   @Override
-  public MapWriter rootAsMap() {
+  public MapWriter rootAsMap(Boolean caseSensitive) {
```

Comment:
I'd suggest that we add instead an optional caseSensitive parameter to the Constructor of ComplexWriterImpl